/who [username] pulls down basic information from gravatar.com #76

nschonni opened the pull request:
Extends the /who command with a user's Gravatar information. There is a Gravatar API limitation where if the email used is not the primary address on the Gravatar account the profile may not show https://secure.gravatar.com/site/implement/profiles/.

nschonni committed:
/who [username] pulls down basic information from gravatar.com #76

davidfowl commented:
Can you update this one?

nschonni committed:
Switched to https for Gravatar profile call

davidfowl commented on JabbR/Chat.js:
```diff
@@ -452,6 +452,13 @@
         }
 
+      $.getJSON('https://secure.gravatar.com/' + userInfo.Hash + '.json?callback=?', function (profile) {
+        ui.addMessage('Gravatar', 'list-header');
```
Any xss issues here?

nschonni replied:
I don't know of any, are you talking about malicious content in the Gravatar profile return? The only parameter (Hash) is generated by JabbR and not user input.
The only one I wouldn't be 100% on is the returned value from the Name portion of the response as that could be set by the user.

samandmoore replied:
So can You test it out with malicious content? And trying to encode it or something?

nschonni committed:
moved markup to jQuery template

@davidfowl requested this to prevent possible xss from the API return
from the manual HTML building

davidfowl commented on JabbR/Chat.js:
```diff
@@ -452,6 +452,10 @@
         }
 
+        $.getJSON('https://secure.gravatar.com/' + userInfo.Hash + '.json?callback=?', function (profile) {
+          ui.showGravatarProfile(profile.entry[0]);
```
4 spaces, no tabs.

nschonni committed:
Changed from 2 spaces to 4

nschonni committed:
Merge remote-tracking branch 'upstream/master' into who-gravatar-profile

Conflicts:
	JabbR/Chat.ui.js

Added trailing comma

nschonni committed:
Fixed merge conflict with closed room changes

nschonni commented:
Something got messed up when I merged with the upstream. Will fix and submit a new PR with a clean push.
